### ISSUE TITLE
Shaka can't play stream manifest with fmp4 chunks if their extension is .ts

### ISSUE THREAD

**Issue by npastuh0v** [shaka-project/shaka-player#7158]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?** -> yes

**If the problem is related to FairPlay, have you read the tutorial?**
 problem not in FairPlay technology

**What version of Shaka Player are you using?** -> 4.9.10

**Can you reproduce the issue with our latest release version?** -> yes

**Can you reproduce the issue with the latest code from `main`?** ->yes

**Are you using the demo app or your own custom app?** ->self application

**If custom app, can you reproduce the issue using our demo app?** ->Haven't tried

**What browser and OS are you using?** ->Google Chrome v.127

Community, good morning. There is a problem with the shaka-player. I remux video to HLS with fmp4 chunk format using the following command:
`ffmpeg -re -i './video.mp4' -c:v copy -c:a copy -f hls -hls_segment_type fmp4 -hls_time 10 -strftime 1 -hls_segment_filename %Y-%m-%d-%H-%M-%S.ts -master_pl_name master.m3u8 media.m3u8`
However, the video does not play, although the hls.js-based player and VLC play it without problems.
When I run the command:
`ffmpeg -re -i './video.mp4' -c:v copy -c:a copy -f hls -hls_segment_type fmp4 -hls_time 10 -strftime 1 -hls_segment_filename %Y-%m-%d-%H-%M-%S.m4s -master_pl_name master.m3u8 media.m3u8` everything plays as it should.

Actually, the question itself: why does the player's performance depend on the file extension? Although, even if it is a route on some web server, I seriously doubt the expediency of this solution. Please correct

**Comment by joeyparrish** [comment]
> Shaka can't play stream manifest with fmp4 chunks if their extension is .ts

> why does the player's performance depend on the file extension?

Because it saves us from making a HEAD request to determine the actual MIME type.

If there's no recognized extension, it takes an additional round-trip to determine the MIME type from the server's `Content-Type` header.

But even if we did that in your case... a typical static content server, in absence of any special configuration, is likely to use a file extension to look up the MIME type.  So the answer would probably be the same.

In short, there's nothing in your example that would allow the player to determine the correct MIME type in advance, which we **must** do to initialize MediaSource.

Native players are different because they typically do everything directly.  They can fetch content and parse it to determine the type before they initialize a demuxer.  On the web, with a separation of duties between JS and the browser's C++ MediaSource implementation, we are supposed to tell it in advance what demuxer to initialize.
